samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt

```diff
@@ -8,62 +8,34 @@ import android.content.Context
 import android.os.Bundle
 import android.support.v7.app.AppCompatActivity
 import android.util.AttributeSet
+import android.util.Log
 import android.view.View
-import kotlinx.android.synthetic.main.activity_main.*
 import mozilla.components.concept.engine.EngineView
-import mozilla.components.feature.session.SessionFeature
 import mozilla.components.feature.session.SessionIntentProcessor
-import mozilla.components.feature.toolbar.ToolbarFeature
 import mozilla.components.support.utils.SafeIntent
 import org.mozilla.samples.browser.ext.components
 
 open class BrowserActivity : AppCompatActivity() {
-    private lateinit var sessionFeature: SessionFeature
-    private lateinit var toolbarFeature: ToolbarFeature
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        toolbar.setMenuBuilder(components.menuBuilder)
-
-        val sessionId = SafeIntent(intent).getStringExtra(SessionIntentProcessor.ACTIVE_SESSION_ID)
-
-        sessionFeature = SessionFeature(
-            components.sessionManager,
-            components.sessionUseCases,
-            engineView,
-            components.sessionStorage,
-            sessionId)
-
-        toolbarFeature = ToolbarFeature(
-            toolbar,
-            components.sessionManager,
-            components.sessionUseCases.loadUrl,
-            components.defaultSearchUseCase,
-            sessionId)
-    }
-
-    override fun onStart() {
-        super.onStart()
-
-        sessionFeature.start()
-        toolbarFeature.start()
-    }
-
-    override fun onStop() {
-        super.onStop()
-
-        sessionFeature.stop()
-        toolbarFeature.stop()
+        if (savedInstanceState == null) {
+            val sessionId = SafeIntent(intent).getStringExtra(SessionIntentProcessor.ACTIVE_SESSION_ID)
+            supportFragmentManager?.beginTransaction()?.apply {
+                replace(R.id.container, BrowserFragment.create(sessionId))
+                commit()
+            }
+        }
     }
 
     override fun onBackPressed() {
-        if (toolbarFeature.handleBackPressed())
-            return
-
-        if (sessionFeature.handleBackPressed())
-            return
+        supportFragmentManager.fragments.forEach {
+            Log.w("SKDBG", "${it.javaClass.canonicalName} is a back handler? ${it is BackHandler}")
```

Comment:
Should we remove the Log call before merging?

Reply:
Lol. Yep!